packages/metal-dom/test/dom.js

```diff
@@ -412,6 +412,92 @@ describe('dom', function() {
 			assert.strictEqual('test', listener.args[0][0].test);
 			document.body.removeChild(element);
 		});
+
+		it('should not trigger dom click event for disabled elements', function() {
+			var input = document.createElement('input');
+			var select = document.createElement('select');
+			var button = document.createElement('button');
+			var textarea = document.createElement('textarea');
+			input.disabled = true;
+			select.disabled = true;
+			button.disabled = true;
+			textarea.disabled = true;
+			document.body.appendChild(input);
+			document.body.appendChild(select);
+			document.body.appendChild(button);
+			document.body.appendChild(textarea);
+
+			var listener1 = sinon.stub();
+			var listener2 = sinon.stub();
+			var listener3 = sinon.stub();
+			var listener4 = sinon.stub();
+
+			input.addEventListener('click', listener1);
+			select.addEventListener('click', listener2);
+			button.addEventListener('click', listener3);
+			textarea.addEventListener('click', listener4);
+
+			dom.triggerEvent(input, 'click');
+			dom.triggerEvent(select, 'click');
+			dom.triggerEvent(button, 'click');
+			dom.triggerEvent(textarea, 'click');
+			assert.strictEqual(0, listener1.callCount);
+			assert.strictEqual(0, listener2.callCount);
+			assert.strictEqual(0, listener3.callCount);
+			assert.strictEqual(0, listener4.callCount);
+		});
+
+		it('should not trigger dom click event to an element which its valid parent is disabled', function() {
+			var parent = document.createElement('fieldset');
+			parent.disabled = true;
+			document.body.appendChild(parent);
+
+			var input = document.createElement('input');
+			var select = document.createElement('select');
+			var button = document.createElement('button');
+			var textarea = document.createElement('textarea');
+			parent.appendChild(input);
+			parent.appendChild(select);
+			parent.appendChild(button);
+			parent.appendChild(textarea);
+
+			var listener1 = sinon.stub();
+			var listener2 = sinon.stub();
+			var listener3 = sinon.stub();
+			var listener4 = sinon.stub();
+
+			input.addEventListener('click', listener1);
+			select.addEventListener('click', listener2);
+			button.addEventListener('click', listener3);
+			textarea.addEventListener('click', listener4);
+
+			dom.triggerEvent(input, 'click');
+			dom.triggerEvent(select, 'click');
+			dom.triggerEvent(button, 'click');
+			assert.strictEqual(0, listener1.callCount);
+			assert.strictEqual(0, listener2.callCount);
+			assert.strictEqual(0, listener3.callCount);
+			assert.strictEqual(0, listener4.callCount);
+		});
+
+		it('should trigger listeners of the click event for non-disabled elements even its parent being disabled', function() {
```

Comment:
The wording here is also a bit weird, it's missing the word `with` after `even`, like:

`should trigger listeners of the click event for non-disabled elements even with its parent being disabled`

Don't worry about it though, we're not native English speakers, we get it wrong from time to time, but it's good to help each other out so we can improve these descriptions whenever we can 😃.